submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/nfs/NfsFilesService.kt

```diff
@@ -1,85 +1,65 @@
-    private val processingService: FileProcessingService,
 ) : FilesService {
-    override fun persistSubmissionFiles(sub: ExtSubmission): ExtSubmission {
-        val subFolder = getOrCreateSubmissionFolder(sub, sub.permissions().folder)
-
-        logger.info { "${sub.accNo} ${sub.owner} Processing files of submission ${sub.accNo} on NFS" }
-        val processed = processAttachedFiles(sub, subFolder, createTempFolder(subFolder, sub.accNo), sub.permissions())
-        logger.info { "${sub.accNo} ${sub.owner} Finished processing files of submission ${sub.accNo} on NFS" }
-        return processed
+    override fun cleanSubmissionFile(file: ExtFile) {
+        val nfsFile = file as NfsFile
+        FileUtils.deleteFile(nfsFile.file)
     }
 
-    override fun cleanSubmissionFiles(sub: ExtSubmission) {
-        logger.info { "${sub.accNo} ${sub.owner} Un-publishing files of submission ${sub.accNo} on NFS" }
-        FileUtils.deleteFile(folderResolver.getSubmissionFtpFolder(sub.relPath).toFile())
-        logger.info { "${sub.accNo} ${sub.owner} Finished un-publishing files of submission ${sub.accNo} on NFS" }
-
-        logger.info { "${sub.accNo} ${sub.owner} Deleting pagetab files of submission ${sub.accNo} on NFS" }
-        sub.allPageTabFiles.filterIsInstance<NfsFile>().forEach { FileUtils.deleteFile(it.file) }
-        logger.info { "${sub.accNo} ${sub.owner} Finished deleting pagetab files of submission ${sub.accNo} on NFS" }
-    }
+    override fun persistSubmissionFile(request: FilePersistenceRequest): ExtFile {
+        val (file, subFolder, targetFolder, permissions) = request as NfsFilePersistenceRequest
+        val target = targetFolder.resolve(file.relPath)
+        val subFile = subFolder.resolve(file.relPath)
 
-    private fun processAttachedFiles(
-        sub: ExtSubmission,
-        subFolder: File,
-        targetFolder: File,
-        permissions: Permissions,
-    ): ExtSubmission {
-        fun copyFile(extFile: NfsFile, idx: Int): NfsFile {
-            val file = extFile.file
-            val target = targetFolder.resolve(extFile.relPath)
-            val subFile = subFolder.resolve(extFile.relPath)
+        if (target.notExist() && subFile.exists() && subFile.md5() == file.md5)
+            moveFile(subFile, target, permissions)
+        else if (target.notExist())
+            copyOrReplaceFile(file.file, target, permissions)
 
-            logger.info { "${sub.accNo} ${sub.owner} Copying file $idx, $file with size ${extFile.size} into $target" }
-
-            if (target.notExist() && subFile.exists() && subFile.md5() == extFile.md5)
-                moveFile(subFile, target, permissions)
-            else if (target.notExist())
-                copyOrReplaceFile(file, target, permissions)
+        return file.copy(fullPath = subFile.absolutePath, file = subFile)
+    }
 
-            return extFile.copy(fullPath = subFile.absolutePath, file = subFile)
-        }
+    fun cleanSubmissionFiles(sub: ExtSubmission) {
```

Comment:
wondering why others methods are an interface that we access using interface while this and getOrCreateSubmissionFolder. It looks super extrange that we comunicate to a class something as a interface (abtracting implementation) and some others using class directly (no abstracting implementation)